### PR TITLE
fix: keep synchronizing the configmaps after websocket disconnect

### PR DESCRIPTION
**Pull request by kamiiiel**
## Issue

https://gravitee.atlassian.net/browse/GKO-54

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ocyyamowty.chromatic.com)

### CODE REVIEW

**Review by jgiovaresco — COMMENTED** [review]
It seems this change is breaking something in the tests... could you have a look?

**Comment by a-cordier** [comment]
Shouldn't we move the retry mechanism to the Kubernetes client in order to focus on connection loss ?

File: gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/kubernetes/fetcher/ConfigMapEventFetcher.java [hunk 1]
```diff
@@ -89,7 +89,8 @@ private Flowable<io.gravitee.kubernetes.client.model.v1.Event<ConfigMap>> watchC
                     .labelSelector(LabelSelector.equals(LABEL_GIO_TYPE, APIDEFINITIONS_TYPE))
                     .build()
             )
-            .retryWhen(errors -> errors.delay(RETRY_DELAY_MILLIS, TimeUnit.MILLISECONDS));
+            .retryWhen(errors -> errors.delay(RETRY_DELAY_MILLIS, TimeUnit.MILLISECONDS))
+            .repeat();
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/src/main/java/io/gravitee/gateway/services/sync/process/kubernetes/fetcher/ConfigMapEventFetcher.java:93]: Can you provide a unit test that reproduces the issue and so, that the fix does the job 🙏?

**Review by jhaeyaert — CHANGES_REQUESTED** [review]

**Comment by jhaeyaert** [comment]
> Shouldn't we move the retry mechanism to the Kubernetes client in order to focus on connection loss ?

@a-cordier I do agree with you. It should probably be fixed on the K8S client so it is transparent for every consumers (at least for this situation where the watch completes by itself without error).

**Comment by a-cordier** [comment]
@jhaeyaert FI the fix has been moved here https://github.com/gravitee-io/gravitee-kubernetes/pull/70

**Comment by a-cordier** [comment]
fixed in https://github.com/gravitee-io/gravitee-kubernetes/commit/b8e7a3de7cfec544538efb00dacc894fa0b80bd8

**Comment by a-cordier** [comment]
fixed in https://github.com/gravitee-io/gravitee-kubernetes/commit/b8e7a3de7cfec544538efb00dacc894fa0b80bd8

**Pull request closed without merging**